pkg/deprecated/ecdsa.go

```diff
@@ -0,0 +1,109 @@
+package deprecated
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/pkg/keys"
+)
+
+func init() {
+	keys.VerifierMap.Delete(data.KeyTypeECDSA_SHA2_P256)
+	keys.VerifierMap.Store(data.KeyTypeECDSA_SHA2_P256, NewDeprecatedEcdsaVerifier)
```

Comment:
IIUC, this isn't race-y because this package depends on the package that happens to add the new verifier to the verifier map.

If that's so, then the `import` statement above is load-bearing; might be worth a comment of warning.

Alternatively (probably better) you could do a check in the other verifier map that stores only if there is no existing entry (I think that's what `LoadOrStore` does).

Either way, a comment about why this is okay would be warranted.

Reply:
Testing suites mess this up, so we can't check that PEM-encoded keys are loaded at first (because each test suite will run init, and the value will be present already).

I added a comment and a check in the deprecation pkg that verifies that we DO override the only PEM verifiers in pkg/keys

Reply:
Nice! I think I'd still *also* like to modify the `pkgs/keys/ecdsa.go:init()` to use `LoadOrStore` instead of `Store`, so we never accidentally replace it.